Character.cpp

```diff
@@ -22,7 +24,39 @@ void Character::attack(Character&  target) const{
 bool Character::isDead() const{
 	return getHp()==0;
 }
+Character Character::parseUnit(const std::string &filename){
+	std::ifstream file;
+    file.open(filename);
+	if (!file.is_open()) throw -99;
+	else{
+		std::string line;
+		std::size_t found;
+		std::string name;
+		int hp,dmg;
+		while (std::getline(file,line)){
+  			if (found=line.find("name")!=std::string::npos){
+				found=line.find(":");
+				line=line.substr(found,line.length());
+				found=line.find("\"");
+				std::size_t found2;
+				found2=line.find_last_of("\"");
+				name=line.substr(found+1,found2-found-1);
+			}
+   			else if (found=line.find("hp")!=std::string::npos){
+				found=line.find(":");
+				hp=std::stoi(line.substr(found+2,line.length()-1));
+			}
+			else if (found=line.find("dmg")!=std::string::npos){
+				found=line.find(":");
+				dmg=std::stoi(line.substr(found+2,line.length()-1));
+			}
+		}
+		file.close();
+		Character characterObj(name,hp, dmg);
+		return characterObj;
```

Comment:
ez celszerubb, ha 1 sorba megy, [RVO](https://en.wikipedia.org/wiki/Copy_elision) valoszinuleg igy is megvan, de atlathatobb is ugy.

Reply:
Javitva, nem masoljuk felesleges a metodusban